### PR TITLE
Feature: Dedicated hosts and host groups

### PR DESCRIPTION
**Pull request by deepaksibm**

### CODE REVIEW

**Comment by deepaksibm** [comment]
Screen Shot 2021-03-15 at 5 21 32 PM
Screen Shot 2021-03-15 at 5 22 10 PM
Screen Shot 2021-03-15 at 5 23 26 PM
Screen Shot 2021-03-15 at 5 25 11 PM
Screen Shot 2021-03-15 at 6 01 56 PM
Screen Shot 2021-03-15 at 6 04 24 PM
Screen Shot 2021-03-15 at 6 06 30 PM
Screen Shot 2021-03-15 at 6 11 37 PM
Screen Shot 2021-03-15 at 6 13 23 PM

**Comment by deepaksibm** [comment]
Terraform Show:

https://ibm.box.com/s/xuf88yxt587s80l8u7wdxhcmlxv1ri6a

**Review by mayukh-sarkar-1 — CHANGES_REQUESTED** [review]

File: website/docs/r/is_dedicated_host.html.markdown [hunk 1]
```diff
@@ -3,7 +3,7 @@ layout: "ibm"
 page_title: "IBM : is_dedicated_host"
 sidebar_current: "docs-ibm-resource-is-dedicated-host"
```
[the remark this answers is not in the document]
Comment (deepaksibm) [reply website/docs/r/is_dedicated_host.html.markdown]: Removed.

File: website/docs/r/is_dedicated_host.html.markdown [hunk 2]
```diff
@@ -3,7 +3,7 @@ layout: "ibm"
 page_title: "IBM : is_dedicated_host"
```
[the remark this answers is not in the document]
Comment (deepaksibm) [reply website/docs/r/is_dedicated_host.html.markdown:4]: Added.

File: ibm/resource_ibm_is_dedicated_host_group_test.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+)
+
+func TestAccIbmIsDedicatedHostGroupBasic(t *testing.T) {
+	var conf vpcv1.DedicatedHostGroup
+	class := "beta"
+	family := "memory"
+	name := fmt.Sprintf("name%d", acctest.RandIntRange(10, 100))
```
Comment (hkantare) [inline ibm/resource_ibm_is_dedicated_host_group_test.go:34]: same as above comment add the prefix tfXXX for the name

File: website/docs/r/is_dedicated_host.html.markdown [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+resource "ibm_is_dedicated_host_group" "dh_group01" {
+  family = "memory"
+  class = "beta"
+  zone = "us-south-1"
+}
+data "ibm_is_dedicated_host_group" "dgroup" {
+	name = ibm_is_dedicated_host_group.dh_group01.name
+}
+resource "ibm_is_dedicated_host" "is_dedicated_host" {
+  profile = "dh2-56x464"
```
Comment (hkantare) [inline website/docs/r/is_dedicated_host.html.markdown:25]: format the example sample code

File: website/docs/r/is_dedicated_host.html.markdown [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+
+* `instance_placement_enabled` - (Optional, bool) If set to true, instances can be placed on this dedicated host.
+* `name` - (Optional, string) The unique user-defined name for this dedicated host. If unspecified, the name will be a hyphenated list of randomly-selected words.
+* `profile` - (Required, string) The Globally unique name of the dedicated host profile to use for this dedicated host.
+* `resource_group` - (Optional, string) The unique identifier of the resource group to use. If unspecified, the account's [default resourcegroup](https://cloud.ibm.com/apidocs/resource-manager#introduction) is used.
+* `host_group` - (Optional, string) The unique identifier of the dedicated host group for this dedicated host.
```
Comment (hkantare) [inline website/docs/r/is_dedicated_host.html.markdown:40]: is the host_group a optional or required?

File: website/docs/r/is_dedicated_host.html.markdown [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+* `id` - The unique identifier of the DedicatedHost.
+* `available_memory` - The amount of memory in gibibytes that is currently available for instances.
+* `available_vcpu` - The available VCPU for the dedicated host.
+* `created_at` - The date and time that the dedicated host was created.
+* `crn` - The CRN for this dedicated host.
+* `host_group` - The unique identifier of the dedicated host group this dedicated host is in.
+* `href` - The URL for this dedicated host.
+* `instance_placement_enabled` - If set to true, instances can be placed on this dedicated host.
+* `instances` - Array of instances that are allocated to this dedicated host.
```
Comment (hkantare) [inline website/docs/r/is_dedicated_host.html.markdown:54]: Give the inner attributes of the instances

File: website/docs/r/is_dedicated_host.html.markdown [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+* `name` - The unique user-defined name for this dedicated host.
+* `profile` - The profile this dedicated host uses.
+* `provisionable` - Indicates whether this dedicated host is available for instance creation.
+* `resource_group` - The unique identifier of the resource group for this dedicated host.
+* `resource_type` - The type of resource referenced.
+* `socket_count` - The total number of sockets for this host.
+* `state` - The administrative state of the dedicated host.The enumerated values for this property are expected to expand in the future. When processing this property, check for and log unknown values. Optionally halt processing and surface the error, or bypass the dedicated host on which the unexpected property value was encountered.
+* `supported_instance_profiles` - Array of instance profiles that can be used by instances placed on this dedicated host.
+* `vcpu` - The total VCPU of the dedicated host.
+* `zone` - The zone this dedicated host resides in.
```
Comment (hkantare) [inline website/docs/r/is_dedicated_host.html.markdown:66]: Add the import section also

File: website/docs/r/is_dedicated_host_group.html.markdown [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+* `id` - The unique identifier of the DedicatedHostGroup.
+* `href` - The URL for this dedicated host group.
+* `crn` - The CRN for this dedicated host group.
+* `created_at` - The date and time that the dedicated host group was created.
+* `dedicated_hosts` - The dedicated hosts that are in this dedicated host group.
+* `name` - The unique user-defined name for this dedicated host group.
+* `resource_type` - The type of resource referenced.
+* `resource_group` - The unique identifier of the resource group for this dedicated host.
+* `supported_instance_profiles` - Array of instance profiles that can be used by instances placed on this dedicated host group.
+* `zone` - The zone this dedicated host resides in.
```
Comment (hkantare) [inline website/docs/r/is_dedicated_host_group.html.markdown:49]: Add the import section

File: ibm/resource_ibm_is_dedicated_host.go [hunk 9]
```diff
@@ -0,0 +1,565 @@
...
+		Schema: map[string]*schema.Schema{
+			"instance_placement_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "If set to true, instances can be placed on this dedicated host.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
```
[the remark this answers is not in the document]
Comment (deepaksibm) [reply ibm/resource_ibm_is_dedicated_host.go]: added validation

File: ibm/resource_ibm_is_dedicated_host.go [hunk 10]
```diff
@@ -0,0 +1,565 @@
...
+
+	createDedicatedHostOptions.SetDedicatedHostPrototype(&dedicatedHostPrototype)
+
+	dedicatedHost, response, err := vpcClient.CreateDedicatedHostWithContext(context, createDedicatedHostOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreateDedicatedHostWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+
+	d.SetId(*dedicatedHost.ID)
```
[the remark this answers is not in the document]
Comment (deepaksibm) [reply ibm/resource_ibm_is_dedicated_host.go:304]: added wait logic

**Pull request merged**